odoo/client.go

```diff
@@ -0,0 +1,108 @@
+package odoo
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+// Client is the base struct that holds information required to talk to Odoo
+type Client struct {
+	baseURL string
+	db      string
+	http    *http.Client
+}
+
+// NewClient returns a new client with its basic fields set
+func NewClient(baseURL, db string) *Client {
+	return &Client{
+		baseURL: strings.TrimSuffix(baseURL, "/"),
+		db:      db,
+		http: &http.Client{
+			Timeout: 10 * time.Second,
+			Jar:     nil, // don't save any cookies!
+		}}
+}
+
+type debugTransport struct {
+	pwRe   *regexp.Regexp
+	logger logr.Logger
+}
+
+func newDebugTransport(logger logr.Logger) *debugTransport {
+	return &debugTransport{
+		pwRe:   regexp.MustCompile(`("password":\s?").+("[,}])`),
+		logger: logger,
+	}
+}
+
+// SetDebugLogger sets the http.Transport field of the internal http client with a transport implementation that logs the raw contents of requests and responses.
+// The log level used is '2'.
+// Any "password":"..." byte content is replaced with a placeholder to avoid leaking credentials.
+// Still, this should not be called in production as other sensitive information might be leaked.
+func (c Client) SetDebugLogger(logger logr.Logger) {
+	c.http.Transport = newDebugTransport(logger)
```

Comment:
newDebugTransport must wrap the original transport if we allow passing a http.Client.